source/adapter.js

```diff
@@ -999,8 +1004,7 @@ if ( navigator.mozGetUserMedia ||
     window.navigator.getUserMedia = getUserMedia;
 
     // Defined mediaDevices when promises are available
-    if ( !navigator.mediaDevices &&
-      typeof Promise !== 'undefined') {
+    if (typeof Promise !== 'undefined') {
```

Comment:
I'm not sure if my fixes will break the original purpose that you have written in here. But my fixes is to ensure that `navigator.mediaDevices` is always defined.